pkg/compression/zstd.go

```diff
@@ -40,13 +48,13 @@ func zstdWriter(dest io.Writer) (io.WriteCloser, error) {
 	return zstd.NewWriter(dest)
 }
 
-func zstdWriterWithLevel(dest io.Writer, level int) (io.WriteCloser, error) {
+func zstdWriterWithLevel(dest io.Writer, level int) (*zstd.Encoder, error) {
 	el := zstd.EncoderLevelFromZstd(level)
 	return zstd.NewWriter(dest, zstd.WithEncoderLevel(el))
 }
 
 // zstdCompressor is a CompressorFunc for the zstd compression algorithm.
-func zstdCompressor(r io.Writer, level *int) (io.WriteCloser, error) {
+func zstdCompressor(r io.Writer, metadata map[string]string, level *int) (io.WriteCloser, error) {
```

Comment:
`metadata` is never used?

Reply:
this is just a preparatory work for https://github.com/containers/storage/pull/775/commits/22a53043dda44ef3f10289c7d2b95ee0dbc3cc65#diff-a32635dc11fd91356384da8c2a47d25c8b278bbd5182eaecd8d2ccaba3d1b670R471-R478